src/native/event_stream_rpc_client.c

```diff
@@ -40,7 +40,8 @@ static void s_destroy_connection_callback_data(struct connection_callback_data *
         return;
     }
 
-    JNIEnv *env = aws_jni_get_thread_env(callback_data->jvm);
+    /********** JNI ENV ACQUIRE **********/
+    JNIEnv *env = aws_jni_acquire_thread_env(callback_data->jvm);
```

Comment:
`s_destroy_connection_callback_data()` is a helper function, not a callback, we could just pass `env` in as an argument